src/bin/common/solving_loop.ml

```diff
@@ -407,8 +410,51 @@ let main () =
       | ":produce-unsat-assumptions"
       | ":print-success"
       | ":random-seed"), _
-      -> Printer.print_wrn "unsupported option %s" name
-    | _ -> Printer.print_wrn "unsupported option %s" name
+      -> unsupported_opt name
+    | _ -> unsupported_opt name
+  in
+
+  let pp_reason_unknown fmt st =
+    let err () =
+      let msg = "Invalid (get-info :reason-unknown)" in
+      Printer.print_err "%s" msg;
+      Format.fprintf fmt "(error %S)" msg
+    in
+    match State.get partial_model_key st with
+    | None ->
+      (* This case should happen only when there is no check-sat before
+         a get-info statement. *)
+      err ()
+    | Some sat ->
+      match SAT.get_unknown_reason sat with
+      | None -> err ()
+      | Some s -> Format.pp_print_string fmt s
+  in
+
+  let handle_get_info (st : State.t) (kind: string) =
+    let print_std =
+      fun (type a) (pp :(Format.formatter -> a -> unit)) (a : a) ->
+        Printer.print_std "(%s %a)" kind pp a
+    in
+    match kind with
+    | ":authors" ->
+      print_std (fun fmt -> Format.fprintf fmt "%S") "Alt-Ergo developers"
+    | ":error-behavior" ->
+      print_std Format.pp_print_string "immediate-exit"
+    | ":name" ->
+      print_std (fun fmt -> Format.fprintf fmt "%S") "alt-ergo"
+    | ":reason-unknown" ->
+      print_std pp_reason_unknown st
+    | ":version" ->
+      print_std Format.pp_print_string Version._version
+    | ":all-statistics"
+    | ":assertion-stack-levels" ->
+      unsupported_opt kind;
+      print_std Format.pp_print_string "unsupported"
+    | _ ->
+      Printer.print_wrn "unknown option %S" kind;
+      print_std Format.pp_print_string "unsupported"
+
   in
```

Comment:
We plan to use `Fmt` and `Logs` everywhere. I prefer `Fmt.pf` rather than `Format.fprintf` and `Fmt.string` instead of `Format.pp_print_string`. I'm ok if you don't apply this modification.